In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.read_csv("C:/Users/Acer/JUPYTER/Assignments/3.Hpyothesis Testing/Costomer+OrderForm.csv")

In [3]:
data.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [4]:
Phillippines_value=data['Phillippines'].value_counts()
Indonesia_value=data['Indonesia'].value_counts()
Malta_value=data['Malta'].value_counts()
India_value=data['India'].value_counts()
print('Phillippines Mean = ',Phillippines_value)
print('Indonesia Mean = ',Indonesia_value)
print('Malta Mean = ',Malta_value)
print('India Mean = ',India_value)

Phillippines Mean =  Error Free    271
Defective      29
Name: Phillippines, dtype: int64
Indonesia Mean =  Error Free    267
Defective      33
Name: Indonesia, dtype: int64
Malta Mean =  Error Free    269
Defective      31
Name: Malta, dtype: int64
India Mean =  Error Free    280
Defective      20
Name: India, dtype: int64


In [5]:
stats.chi2_contingency([data['Phillippines'].value_counts(), 
                        data['Indonesia'].value_counts(), 
                        data['Malta'].value_counts(), 
                        data['India'].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

 P value of Chi-Square test  == 0.277 > α
 null hypothesis is Accepted
 Thus HO is accepted.  
 The defective % does not varies significantly by centres